parse/Parse.cpp

```diff
@@ -234,9 +235,9 @@ namespace parse {
     const sregex MACRO_KEY = +_w;   // word character (alnum | _), one or more times, greedy
     const sregex MACRO_TEXT = -*_;  // any character, zero or more times, not greedy
     const sregex MACRO_DEFINITION = (s1 = MACRO_KEY) >> _n >> "'''" >> (s2 = MACRO_TEXT) >> "'''" >> _n;
-    const sregex MACRO_INSERTION = "[[" >> *space >> (s1 = MACRO_KEY) >> *space >> "]]";
+    const sregex MACRO_INSERTION = "[[" >> *space >> (s1 = MACRO_KEY) >> *space >> !("(" >> (s2 = +~_n) >> ")") >> "]]";
```

Comment:
I think this MACRO_INSERTION definition could use some improvement.  For our most typical use case, where the macro is used by itself on a line, as with the Xeno example, this looks fine to me.  But if you were using it in the middle of some other text, it seems it could be at the the least horribly inefficient, but I think unless I'm mistaken, actually problematic.  Consider the basic text
```
Ok this is some entry with a first macro useage [[ANY_MACRO(param1)]].  Then on the same line we also have [[ANY_MACRO(param2)]] and maybe a little more text.
```

When it hits that first ANY_MACRO, isn't s2 going to be all of 
```
param1)]].  Then on the same line we also have [[ANY_MACRO(param2
```
?

It seems to me that rather than simply "+~_n]" that you'd also want to exclude ")" as well-- it's not like anyone really *needs* to use a parenthesis as part of their macro substitution key.